pkg/cluster/config/v1alpha2/types.go

```diff
@@ -42,6 +42,12 @@ type Node struct {
 	// Role defines the role of the nodw in the in the Kubernetes cluster managed by `kind`
 	// Defaults to "control-plane"
 	Role NodeRole `json:"role,omitempty"`
+	// CNI is the name of the desired network plugin
+	CNI string `json:"cni,omitempty"`
```

Comment:
are we going to put different CNI per node?

Reply:
That's a good point. 
I don't think that's a very likely scenario, but at the same time I don't believe there is any strict requirement that all nodes in a cluster use the same CNI plugin. For example, a particular set of nodes may have specialized NICs for which a different CNI plugin may be desirable.

Reply:
we should definitely move the CNI object outside of the Node and into the main Config object.
CNI should be per cluster.

Reply:
@neolit123 CNIConfig can contain node specific data. For example, the host-local ipam consists of fields (e.g. subnet, gateway, routes) that are node specific (https://github.com/containernetworking/plugins/tree/master/plugins/ipam/host-local) 
I suspect that this will not play well with Replicas since nodes will not be exact copies of each other. However, this is something that can be documented.

Reply:
> For example, a particular set of nodes may have specialized NICs for which a different CNI plugin may be desirable.

But these are not on real NICs.

I also think having CNI config is perhaps not the right level of complexity for kind. Most CNI are deployable as a k8s manifest, I think we can support that or not installing a CNI at all and leaving it to the end user. Ref https://github.com/kubernetes-sigs/kind/issues/278